fe/fe-core/src/main/java/org/apache/doris/planner/EsScanNode.java

```diff
@@ -257,11 +257,13 @@ private List<TScanRangeLocations> getShardLocations() throws UserException {
                 // Generate on es scan range
                 TEsScanRange esScanRange = new TEsScanRange();
                 esScanRange.setEsHosts(shardAllocations);
-                esScanRange.setIndex(shardRouting.get(0).getIndexName());
+                if (shardRouting.size() > 0 && shardRouting != null) {
```

Comment:
Will this cause errors in query?

Reply:
The query will appear: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0

Reply:
I mean is whether not to set index will cause subsequent query failure.

Reply:
I confirm